packages/blockchain-wallet-v4-frontend/src/data/components/dex/sagas.ts

```diff
@@ -12,6 +12,34 @@ import { DexChain, DexChainList, DexChainTokenList, DexSwapForm } from './types'
 const { DEX_SWAP_FORM } = model.components.dex
 
 export default ({ api }: { api: APIType }) => {
+  const fetchUserEligibility = function* () {
+    try {
+      // TODO: since MVP only supports ETH chain
+      const token = 'ETH'
+      const state = yield select()
+
+      const nonCustodialCoinAccounts: Record<string, SwapAccountType[]> = yield select(() =>
+        selectors.coins.getCoinAccounts(state, {
+          coins: [token],
+          nonCustodialAccounts: true
+        })
+      )
+
+      const walletAddress = nonCustodialCoinAccounts[token][0].address
+      if (!walletAddress) {
+        return Remote.Failure('No user wallet address')
```

Comment:
Did you mean to dispatch an action here?

Reply:
Fixed in the next PR